ethcore/transaction/src/transaction/eip2930.rs

```diff
@@ -0,0 +1,240 @@
+    /// Transaction data.
+    pub(crate) data: Bytes,
+    /// Access list.
+    pub(crate) access_list: AccessList,
+}
+
+impl Eip2930Transaction {
+    const fn payload_size(&self) -> usize { 8 }
+
+    /// Append object with a without signature into RLP stream
+    fn rlp_append_unsigned_transaction(&self, s: &mut RlpStream) {
+        s.append(&(TxType::Type1 as u8));
+        s.begin_list(self.payload_size());
+        s.append(&self.chain_id);
+        s.append(&self.nonce);
+        s.append(&self.gas_price);
+        s.append(&self.gas);
+        s.append(&self.action);
+        s.append(&self.value);
+        s.append(&self.data);
+        s.append(&self.access_list);
+    }
+}
+
+impl TransactionShared for Eip2930Transaction {
+    fn nonce(&self) -> U256 { self.nonce }
+    fn action(&self) -> &Action { &self.action }
+    fn value(&self) -> U256 { self.value }
+    fn data(&self) -> &Bytes { &self.data }
+    /// The message hash of the transaction.
+    fn message_hash(&self, _chain_id: Option<u64>) -> H256 {
+        let mut stream = RlpStream::new();
+        self.rlp_append_unsigned_transaction(&mut stream);
+        keccak(stream.as_raw())
+    }
+}
+
+impl rlp::Decodable for Eip2930Transaction {
+    fn decode(d: &Rlp) -> Result<Self, DecoderError> {
+        if d.as_raw().len() < 2 {
+            return Err(DecoderError::RlpIsTooShort);
+        }
+        let version: u8 = d.as_raw()[0];
+        if TxType::from(version) != TxType::Type1 {
+            return Err(DecoderError::Custom("bad tx version"));
+        }
+        let list = Rlp::new(&d.as_raw()[1..]);
+        Ok(Eip2930Transaction {
+            chain_id: list.val_at(0)?,
+            nonce: list.val_at(1)?,
+            gas_price: list.val_at(2)?,
+            gas: list.val_at(3)?,
+            action: list.val_at(4)?,
+            value: list.val_at(5)?,
+            data: list.val_at(6)?,
+            access_list: list.val_at(7)?,
+        })
+    }
+}
+
+/// Signed transaction information without verified signature.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct UnverifiedEip2930Transaction {
+    /// Plain Transaction.
+    unsigned: Eip2930Transaction,
+    /// The V field of the signature
+    v: u64,
+    /// The R field of the signature; helps describe the point on the curve.
+    r: U256,
+    /// The S field of the signature; helps describe the point on the curve.
+    s: U256,
+    /// Hash of the transaction
+    hash: H256,
+}
+
+impl Deref for UnverifiedEip2930Transaction {
+    type Target = Eip2930Transaction;
+
+    fn deref(&self) -> &Self::Target { &self.unsigned }
+}
+
+impl rlp::Decodable for UnverifiedEip2930Transaction {
+    fn decode(d: &Rlp) -> Result<Self, DecoderError> {
+        let unsigned = Eip2930Transaction::decode(d)?;
+        let hash = keccak(d.as_raw());
+        let offset = unsigned.payload_size();
+        if d.as_raw().is_empty() {
+            return Err(DecoderError::RlpIsTooShort);
+        }
```

Comment:
Same here, this is checked as part of E`ip2930Transaction::decode(d)`

Reply:
fixed in [5dbe940](https://github.com/KomodoPlatform/mm2-parity-ethereum/pull/1/commits/5dbe94081c38472bd733c15a1b4f614e6dcc54dd)